### PR TITLE
net-p2p/i2p: bump to 0.9.26

### PR DESCRIPTION
**Pull request by tharvik**
From [gbugs#586850](https://bugs.gentoo.org/show_bug.cgi?id=586850).

* bump to 0.9.26
* remove 0.9.24
* remove some stall files

Github seems to have some issues thinking it is a renaming.

### CODE REVIEW

**Comment by gktrk** [comment]
@tomboy-64 @gentoo/java @gentoo/proxy-maint

File: net-p2p/i2p/i2p-0.9.26.ebuild [hunk 1]
```diff
@@ -72,15 +75,16 @@ src_prepare() {
 	if ! use nls; then
 		echo "require.gettext=false" >> override.properties
```
Comment (SoapGentoo) [inline net-p2p/i2p/i2p-0.9.26.ebuild]: `echo`'ing into a file requires `|| die` as I/O errors could occur

File: net-p2p/i2p/i2p-0.9.26.ebuild [hunk 2]
```diff
@@ -72,15 +75,16 @@ src_prepare() {
...
 	fi
+	eapply_user
 }
 
 src_install() {
+	# add libs
+	epatch "${FILESDIR}/${P}-add_libs.patch"
```
Comment (SoapGentoo) [inline net-p2p/i2p/i2p-0.9.26.ebuild]: don't mix patching into different phase functions. Patching should happen in `src_prepare`, preferably by adding the patches into the global `PATCHES` array variable.

File: net-p2p/i2p/i2p-0.9.26.ebuild [hunk 3]
```diff
@@ -72,15 +75,16 @@ src_prepare() {
 	if ! use nls; then
 		echo "require.gettext=false" >> override.properties
 	fi
+	eapply_user
```
Comment (SoapGentoo) [inline net-p2p/i2p/i2p-0.9.26.ebuild]: rather use `default` here, to also apply other patches.

File: net-p2p/i2p/i2p-0.9.26.ebuild [hunk 4]
```diff
@@ -93,14 +97,13 @@ src_install() {
...
 
-	# This is ugly, but to satisfy all non-system .jar dependencies, jetty and
-	# systray4j would need to be packaged. The former would be too large a task
-	# for an unseasoned developer and systray4j hasn't been touched in over 10
-	# years. This seems to be the most pragmatic solution
+	# This is ugly, but to satisfy all non-system .jar dependencies, jetty
+	# would need to be packaged. It would be too large a task
+	# for an unseasoned developer. This seems to be the most pragmatic solution
 	java-pkg_jarinto "${I2P_ROOT}/lib"
 	for i in BOB commons-el commons-logging i2p i2psnark i2ptunnel \
```
Comment (SoapGentoo) [inline net-p2p/i2p/i2p-0.9.26.ebuild]: declare `i` as local variable before

**Comment by tharvik** [comment]
@SoapGentoo thanks for the feedback, here goes the updated commit.

**Comment by tomboy-64** [comment]
examined it and ran it over the weekend (amd64 and arm). looks fine to me.
thanks for cleaning it up.

**Comment by chewi** [comment]
This isn't new but can someone please explain the contents of RDEPEND? All those variations of icedtea aren't making sense to me.

**Comment by tharvik** [comment]
@chewi it is because when i2p goes full ECDSA, we can simply remove some lines to have `[nss,-sunec]` flag (base on [ECDSA help page of i2p](https://trac.i2p2.de/wiki/Crypto/ECDSA)), but as it is not mandatory yet, we don't want to force it. Also, we have to explicit every JRE as there is no USE flag for `virtual/jre`.
I guess we can remove some now, as icedtea:6 is gone from tree.

**Comment by chewi** [comment]
Okay, it's a bit weird but I guess that makes sense. The references to 6 have gone already.

**Comment by tharvik** [comment]
FYI, I've amend it to use a `keepdir` for the config dir, to fix [gbugs#588526](https://bugs.gentoo.org/show_bug.cgi?id=588526).

**Pull request closed without merging**